src/GafferUI/Handle.cpp

```diff
@@ -185,6 +185,11 @@ Imath::V3f Handle::rasterScaleFactor() const
 	return s1 * s2;
 }
 
+float Handle::transformationFactor() const
```

Comment:
This seemed a bit off to me, but I'm not sure I can articulate why. I think perhaps because other than this, the base class delegates all details of the drag behaviour entirely to the derived classes?

I do have a slightly less hand-wavey justification though : I'm finding I might want to let go of Shift during the drag, and go back to regular speed. I think this means it is easier if `TranslateHandle::translation()`, `ScaleHandle::scaling()` and `RotateHandle::rotation()` just read the current modifier state from their `event` argument. That way we don't need any additional state on the base class - all the state is in the `DragDropEvent`.

There's perhaps also an argument for making it possible to go fast for a bit, and then slow down a bit (without snapping back to the position you'd be in if you'd gone slow from the start), which would require pushing more state into the PlanarDrag and LinearDrag helpers (and probably introducing an AngularDrag helper too). I'm happy to leave that as future work, but ditching `Handle::transformationFactor()` would be a necessary prerequisite, so I suggest we ditch in now.

Reply:
I'll resolve this one myself this time as it's not up to date anymore and the new approach needs reviewing.